### DISCUSSION TITLE
How to use a specific certificate chain?

### DISCUSSION THREAD

**Discussion by wsw70** [go-resty/resty#634] [Q&A] [answered]
I have internal sites that present certificates signed by an internal PKI.
I have exported the **chain of certificates** as PEM and get a singe file with a few of them:

```
-----BEGIN CERTIFICATE-----
MIIIITCCBgmgAwIBAgITGQAHr8UnpOy1chtuhgACAAevxTANBgkqhkiG9w0BAQsF
(...)
YsYj1UL+pObfq4z784YMnSRq6UHp
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIIBTCCBe2gAwIBAgITLwAAAJSPEqECPjb0QQABAAAAlDANBgkqhkiG9w0BAQsF
(...)
20hQhjVQN3D5hywcA8DisJw8nO21M06vq7mUWetUwg/qPCahUBquQpQ=
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIFyDCCA7CgAwIBAgIQVUpR4DkISZZFMn5JOGyAYTANBgkqhkiG9w0BAQsFADBX
(...)
rtWNw7yi3HveIo0uq36IWMgg4vfMRJ+lxlCP63f2uHGDjY/eWD0lVp8oESM=
-----END CERTIFICATE-----
```

I tried to use this file as 

```go
r, err := resty.New().
	SetRootCertificate("mycerts.pem").
	R().
	SetResult(&u).
	SetHeader("Accept", "application/json").
	Get(fmt.Sprintf("%v/id/%v", whoUrl, user.EmployeeNumber))
```

This call fails with a `tls: failed to verify certificate: x509: certificate signed by unknown authority`.

I also tried to break the certificates into separate files (three in my case) and chain `SetRootCertificate`:

```
r, err := resty.New().
	SetRootCertificate("cert-1.pem").
	SetRootCertificate("cert-2.pem").
	SetRootCertificate("cert-3.pem").
	R().
(...)
```

Should I declare this chain in another way?

**Comment by elfranne** [comment, marked as answer]
Have a look here: 
https://pkg.go.dev/github.com/go-resty/resty/v2#Client.SetTLSClientConfig
https://pkg.go.dev/crypto/tls#Dial
